hazelcast-tpc-engine/src/main/java/com/hazelcast/internal/tpc/nio/NioEventloop.java

```diff
@@ -36,57 +32,24 @@
 class NioEventloop extends Eventloop {
 
     final Selector selector = SelectorOptimizer.newSelector();
-    private final NioReactor reactor;
 
     NioEventloop(NioReactor reactor, NioReactorBuilder builder) {
         super(reactor, builder);
-        this.reactor = reactor;
     }
 
     @Override
     protected void run() throws Exception {
-        final NanoClock nanoClock = this.nanoClock;
-        final boolean spin = this.spin;
-        final Selector selector = this.selector;
-        final AtomicBoolean wakeupNeeded = this.wakeupNeeded;
-        final MpmcArrayQueue externalTaskQueue = this.externalTaskQueue;
-        final Scheduler scheduler = this.scheduler;
-
         boolean moreWork = false;
         do {
             int keyCount;
             if (spin || moreWork) {
                 keyCount = selector.selectNow();
             } else {
-                wakeupNeeded.set(true);
-                if (externalTaskQueue.isEmpty()) {
-                    if (earliestDeadlineNanos == -1) {
-                        keyCount = selector.select();
-                    } else {
-                        long timeoutMillis = NANOSECONDS.toMillis(earliestDeadlineNanos - nanoClock.nanoTime());
-                        keyCount = timeoutMillis <= 0
-                                ? selector.selectNow()
-                                : selector.select(timeoutMillis);
-                    }
-                } else {
-                    keyCount = selector.selectNow();
-                }
-                wakeupNeeded.set(false);
+                keyCount = select();
```

Comment:
This is not acceptable to me. I need to have the content of this code on a single screen so I can understand what is going on. So please restore the code of the loop to its original state.